doc/nrf/ug_thread_architectures.rst

```diff
@@ -204,5 +207,173 @@ It also has the following disadvantages:
+
+.. _ug_thread_architectures_communication_rxtx_rx:
+
+Receiving data and RX data flows
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+The section illustrates the RX data flows for UART and SPI for when the commands are received by NCP/RCP:
+
+* Data RX flow for UART
+
+  .. figure:: /images/thread_data_flow_rx_uart.svg
+     :alt: Data RX flow for UART
+
+     Data RX flow for UART
+
+  In this flow:
+
+  1. UART interface stores up to 6 bytes in the hardware FIFO.
+  #. HDLC-encoded data is stored in the Driver receive buffer.
+  #. HDLC data is decoded and stored in the NCP UART Driver receive buffer.
+  #. Spinel commands are dispatched and handled by proper routines.
+
+     * If a command requires a response, it will be added to the NCP response queue for later execution.
+
+* Data RX flow for SPI
+
+  .. figure:: /images/thread_data_flow_rx_spi.svg
+     :alt: Data RX flow for SPI
+
+     Data RX flow for SPI
+
+  In this flow:
+
+  1. SPI interface saves data into the NCP SPI RX buffer.
+  #. NCP obtains pointer to the Spinel frame in the buffer and handles it.
+
+     * If a command requires a response, it will be added to the NCP response queue for later execution.
+
+
+.. _ug_thread_architectures_communication_rxtx_tx:
+
+Transmitting data
+~~~~~~~~~~~~~~~~~
+
+NCP/RCP has the following process for sending responses:
+
+1. After a command is received, the response ends up in the NCP/RCP Response Queue.
+#. In the NCP/RCP Response Queue, the command is checked for the data required by the host.
+#. NCP/RCP gathers the data and writes the response to the TX buffer by emptying the NCP/RCP Response Queue.
+
+   * The process of writing the frames to the buffer is described in the :ref:`Writing to the buffer paragraph <ug_thread_writing_buffer>`.
+
+#. NCP/RCP sends the response from the TX buffer to the host.
+
+.. _ug_thread_update_commands:
+
+Unsolicited update commands
+~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+The Spinel also allows sending unsolicited update commands from NCP to the host, for example when NCP or a node receives a IPv6 packet that must be forwarded to the host.
+
+The unsolicited update commands have the following characteristics:
+
+* They are written to the TX buffer.
+* They are asynchronous.
+* All have the TID value equal to zero.
+* They have low priority.
+
+The unsolicited update commands include both delayable and droppable frames (see :ref:`ug_thread_architectures_communication_priorities`), prioritized in the following order:
+
+1. Delayable frames:
+
+   1. MAC, IPv6 and UDP forwarding stream properties.
+   #. Property value notification commands, including last status update.
+
+#. Droppable frames:
+
+   1. Debug stream for application.
+
+      * This is a separate log for application that has a property ID field that allows the application to distinguish different debug streams.
+
+   #. Log.
+
+      * This is a log that can be used to report errors in the OpenThread stack to the host.
```

Comment:
There is a change (already merged into Zephyr, but not in NCS yet), which allows to report logs from OpenThread as well as from Zephyr system using Spinel. I don't know if this doc considers using Zephyr on NCP, but if yes, I would remind it, as it's quite useful feature.